dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ArgumentTypesResolver.java

```diff
@@ -97,10 +97,17 @@ private Optional<ArgumentType> getPrimitiveType(TypeMirror typeMirror) {
 
     private Optional<ArgumentType> getListType(TypeMirror typeMirror) {
         TypeName typeName = TypeName.get(typeMirror);
+        Optional<TypeMirror> innerTypeMirror = context.getGenericInnerType(List.class, typeMirror);
 
-        return context.getGenericInnerType(List.class, typeMirror)
-                .flatMap(this::getPrimitiveSerializerMethodName)
-                .map(methodName -> ArgumentTypes.list(typeName, methodName));
+        return innerTypeMirror
```

Comment:
nit: Inline the `getGenericInnerType` call - `innerTypeMirror` is only used here

```suggestion
        return context.getGenericInnerType(List.class, typeMirror)
```

Reply:
Fixed